include/alp/reference/matrix.hpp

```diff
@@ -432,6 +439,25 @@ namespace alp {
 				std::pair< size_t, size_t > dims() const noexcept {
 					return static_cast< const DerivedMatrix * >( this )->_dims();
 				}
+
+				template< typename MatrixType >
+				friend const typename MatrixType::access_type access( const MatrixType &A, const size_t i, const size_t j );
+
+				template< typename MatrixType >
+				friend typename MatrixType::access_type access( MatrixType &A, const size_t i, const size_t j );
+
+				template< typename AccessType >
+				const AccessType access( const size_t i, const size_t j ) const {
```

Comment:
Should we go directly for `DerivedMatrix::access_type` here instead of generalizing with `AccessType`?